buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/apiinfo/ApiInformationTask.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins.apiinfo;
+
+import groovy.transform.CompileStatic;
+import jdk.nashorn.internal.objects.annotations.Property;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+    Task generates the api diff of the current source code against the api.txt file stored
+    alongside the project's src directory.
+ */
+@CompileStatic
+public class ApiInformationTask extends DefaultTask {
+
+    @Property
+    String metalavaBinaryPath;
+
+    @Property
+    String apiTxt;
+
+    @Property
+    String sourcePath;
+
+    @Property
+    String baselinePath;
+
+    @Property
+    boolean updateBaseline;
+
+    @Property
+    String outputPath;
+
+
+    private String getCmdOutput (String cmdToRun) {
+        ProcessBuilder processBuilder = new ProcessBuilder(cmdToRun.split(" "));
```

Comment:
you should be able to just use `getProject().javaexec(spec -> {/*configure javaexec*/});`

Reply:
or alternatively just plain `getProject().exec()`

Reply:
this is still not done